### ISSUE TITLE
Bugfix: docker-compose instructions create a sonic container that fails to start

### ISSUE THREAD

**Issue by JohnMaguire** [ArchiveBox/ArchiveBox#621]
#### Describe the bug

I followed the docker-compose instructions from the README. This is the result:

```
[root@Acheron archivebox]# docker-compose ps
         Name                        Command                State             Ports
--------------------------------------------------------------------------------------------
archivebox_archivebox_1   dumb-init -- /app/bin/dock ...   Up         0.0.0.0:8000->8000/tcp
archivebox_sonic_1        sonic -c /etc/sonic.cfg          Exit 101
```
```
[root@Acheron archivebox]# docker-compose logs sonic
Attaching to archivebox_sonic_1
sonic_1       | thread 'main' panicked at 'cannot read config file: Os { code: 21, kind: Other, message: "Is a directory" }', src/config/reader.rs:24:14
sonic_1       | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
sonic_1       | thread 'main' panicked at 'cannot read config file: Os { code: 21, kind: Other, message: "Is a directory" }', src/config/reader.rs:24:14
sonic_1       | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Search seems to work anyway.

I would expect one of:

a. `sonic` container is not created by default if it requires the user to manually create a config and is not necessary to run ArchiveBox
b. `config.cfg` is created for me by the init script, using the environment variable I set in the docker-compose file
c. `config.cfg` is not required by sonic (however, this is not the case: https://github.com/valeriansaliou/sonic/issues/197)

#### Steps to reproduce

From the README:

```
# create a new empty directory and initalize your collection (can be anywhere)
mkdir ~/archivebox && cd ~/archivebox
curl -O https://raw.githubusercontent.com/ArchiveBox/ArchiveBox/master/docker-compose.yml
docker-compose run archivebox init
docker-compose run archivebox --version

# start the webserver and open the UI (optional)
docker-compose run archivebox manage createsuperuser
docker-compose up -d
open http://127.0.0.1:8000

# you can also add links and manage your archive via the CLI:
docker-compose run archivebox add 'https://example.com'
docker-compose run archivebox status
docker-compose run archivebox help  # to see more options
```

#### ArchiveBox version
```
[root@Acheron archivebox]# docker-compose run archivebox --version
Starting archivebox_sonic_1 ... done
Creating archivebox_archivebox_run ... done
ArchiveBox v0.5.3
Cpython Linux Linux-5.9.1-arch1-1-x86_64-with-glibc2.28 x86_64 (in Docker)

[i] Dependency versions:
 √  ARCHIVEBOX_BINARY     v0.5.3          valid     /usr/local/bin/archivebox
 √  PYTHON_BINARY         v3.9.1          valid     /usr/local/bin/python3.9
 √  DJANGO_BINARY         v3.1.3          valid     /usr/local/lib/python3.9/site-packages/django/bin/django-admin.py
 √  CURL_BINARY           v7.64.0         valid     /usr/bin/curl
 √  WGET_BINARY           v1.20.1         valid     /usr/bin/wget
 √  NODE_BINARY           v15.5.1         valid     /usr/bin/node
 √  SINGLEFILE_BINARY     v0.1.14         valid     /node/node_modules/single-file/cli/single-file
 √  READABILITY_BINARY    v0.1.0          valid     /node/node_modules/readability-extractor/readability-extractor
 √  MERCURY_BINARY        v1.0.0          valid     /node/node_modules/@postlight/mercury-parser/cli.js
 √  GIT_BINARY            v2.20.1         valid     /usr/bin/git
 √  YOUTUBEDL_BINARY      v2021.01.03     valid     /usr/local/bin/youtube-dl
 √  CHROME_BINARY         v87.0.4280.88   valid     /usr/bin/chromium
 √  RIPGREP_BINARY        v0.10.0         valid     /usr/bin/rg

[i] Source-code locations:
 √  PACKAGE_DIR           22 files        valid     /app/archivebox
 √  TEMPLATES_DIR         3 files         valid     /app/archivebox/themes

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled
 -  COOKIES_FILE          -               disabled

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data
 √  SOURCES_DIR           1 files         valid     ./sources
 √  LOGS_DIR              0 files         valid     ./logs
 √  ARCHIVE_DIR           1 files         valid     ./archive
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf
 √  SQL_INDEX             204.0 KB        valid     ./index.sqlite3
```

```
[root@Acheron archivebox]# docker version
Client:
 Version:           20.10.2
 API version:       1.40
 Go version:        go1.15.6
 Git commit:        2291f610ae
 Built:             Tue Jan  5 19:56:21 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          19.03.13-ce
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.15.2
  Git commit:       4484c46d9d
  Built:            Sat Sep 26 12:03:35 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.1.m
  GitCommit:        c623d1b36f09f8ef6536a057bd658b3aa8632828.m
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

```
[root@Acheron archivebox]# docker-compose version
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.0l  10 Sep 2019
```

**Comment by JohnMaguire** [comment]
After deleting the automatically generated directory and copying https://github.com/valeriansaliou/sonic/blob/master/config.cfg to `./etc/sonic/config.cfg`, I get the following error:

```
[root@Acheron archivebox]# docker-compose up -d
Starting archivebox_sonic_1 ... error

ERROR: for archivebox_sonic_1  Cannot start service sonic: OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: rootfs_linux.go:59: mounting "/opt/docker/projects/archivebox/etc/sonic/config.cfg" to rootfs at "/var/lib/docker/overlay2/13cc28834c925d7a3565efeb8f3b487349a039735d495dd48e1817cc67f39e57/merged/etc/sonic.cfg" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type

ERROR: for sonic  Cannot start service sonic: OCI runtime create failed: container_linux.go:370: starting container process caused: process_linux.go:459: container init caused: rootfs_linux.go:59: mounting "/opt/docker/projects/archivebox/etc/sonic/config.cfg" to rootfs at "/var/lib/docker/overlay2/13cc28834c925d7a3565efeb8f3b487349a039735d495dd48e1817cc67f39e57/merged/etc/sonic.cfg" caused: not a directory: unknown: Are you trying to mount a directory onto a file (or vice-versa)? Check if the specified host path exists and is the expected type
ERROR: Encountered errors while bringing up the project.
```

**Comment by pirate** [comment]
Thanks for reporting, @jdcaballerov are there any steps missing I need to add to the readme to get Sonic working?

**Comment by JohnMaguire** [comment]
One option might be to simply include the example `config.cfg` from the `sonic` repo when running the init command.

Sorry to continue commenting, but I'm working through this still... now that Sonic is running, search doesn't seem to return any results. After `docker-compose stop sonic`, I get results again, but with the notice "Error from the search backend, only showing results from default admin search fields - Error: [Errno -2] Name or service not known" Not sure if I need to kick off an index job?

**Comment by jdcaballerov** [comment]
@pirate @JohnMaguire  The required `config,cfg` is [included in](https://github.com/ArchiveBox/ArchiveBox/tree/dev/etc/sonic) `etc/` we need to devise a solution to include it for the people not having the `etc` directory ( the config file is mounted as a volume) and are using an image from docker registry. (BUG)

Using the config from https://github.com/ArchiveBox/ArchiveBox/tree/dev/etc/sonic will likely err since it uses IPV6 not enabled by default in docker, as I remember. There are other parameters tuned for this use case.

**Comment by JohnMaguire** [comment]
Perhaps a simple solution would be to amend the docker-compose instructions to include:

```
mkdir -p ~/archivebox{,etc/sonic/} && cd ~/archivebox/  # in lieu of mkdir ~/archivebox && cd ~/archivebox, or as a discrete step
curl -O https://raw.githubusercontent.com/ArchiveBox/ArchiveBox/dev/etc/sonic/config.cfg > etc/sonic/config.cfg
```

**Comment by JohnMaguire** [comment]
After updating to use the config from this repo, I am still not seeing results however...

Pre-Search:

<img width="1625" alt="Screen Shot 2021-01-19 at 10 52 32 PM" src="https://user-images.githubusercontent.com/440033/105124893-1d26ff80-5aa9-11eb-8e0d-edeed7ca8301.png">

Post-Search:

<img width="1610" alt="Screen Shot 2021-01-19 at 10 52 40 PM" src="https://user-images.githubusercontent.com/440033/105124892-1d26ff80-5aa9-11eb-93ce-7387bb71b851.png">

**Comment by jdcaballerov** [comment]
@JohnMaguire Sonic will index the newly added links. For old links you need to manually trigger indexing  with `update` index only option.

**Comment by JohnMaguire** [comment]
```
[root@Acheron archivebox]# docker-compose run archivebox update --index-only
Creating archivebox_archivebox_run ... done
[i] [2021-01-20 04:01:36] ArchiveBox v0.5.3: archivebox update --index-only
    > /data

[*] Indexing url: https://google.com/ in the search index
```

But still no results or logs to note (tried `google`, `google.com` and `Google`):

```
archivebox_1  | [20/Jan/2021 04:02:33] "GET /admin/core/snapshot/?q=google.com HTTP/1.1" 200 11806
archivebox_1  | [20/Jan/2021 04:02:33] "GET /admin/jsi18n/ HTTP/1.1" 200 3321
sonic_1       | (INFO) - took 0ms/122us/122683ns to process channel message
sonic_1       | (INFO) - stop search executor as no iid was found in common for term: google
sonic_1       | (INFO) - got search executor final oids: []
sonic_1       | (INFO) - took 1ms/1579us/1579322ns to process channel message
sonic_1       | (INFO) - took 0ms/58us/58790ns to process channel message
archivebox_1  | [20/Jan/2021 04:02:35] "GET /admin/core/snapshot/?q=Google HTTP/1.1" 200 11738
archivebox_1  | [20/Jan/2021 04:02:35] "GET /admin/jsi18n/ HTTP/1.1" 200 3321
sonic_1       | (INFO) - took 0ms/119us/119481ns to process channel message
sonic_1       | (INFO) - stop search executor as no iid was found in common for term: google
sonic_1       | (INFO) - got search executor final oids: []
sonic_1       | (INFO) - took 1ms/1692us/1692223ns to process channel message
sonic_1       | (INFO) - took 0ms/64us/64081ns to process channel message
archivebox_1  | [20/Jan/2021 04:02:38] "GET /admin/core/snapshot/?q=google HTTP/1.1" 200 11738
archivebox_1  | [20/Jan/2021 04:02:38] "GET /admin/jsi18n/ HTTP/1.1" 200 3321
sonic_1       | (INFO) - done scanning for kv store pool items to janitor, expired 0 items, now has 1 items
sonic_1       | (INFO) - done scanning for fst store pool items to janitor, expired 0 items, now has 1 items
sonic_1       | (INFO) - no kv store pool items need to be flushed at the moment
sonic_1       | (INFO) - no fst store pool items need to consolidate at the moment
sonic_1       | (INFO) - ran tasker tick (took 0s + 0ms)
```

**Comment by JohnMaguire** [comment]
OK, searching on the public index works correctly. From the admin UI, it returns no results.

**Comment by jdcaballerov** [comment]
@JohnMaguire rebuilding the index is a task that is managed by sonic and doesn't occur immediately after being instructed to do so. Allow some time without killing it and let us know.

**Comment by JohnMaguire** [comment]
I added example.com as well, and am seeing the same behavior:

- Searching for "example" on http://192.168.1.14:8000/public/?q=example works.
- Searching for "example" on http://192.168.1.14:8000/admin/core/snapshot/grid/?q=example returns no results.

I know you mentioned "Sonic will only index the newly added links after it's enabled," so I think it should index this? And since the public search is returning, it seems unlikely that indexing is broken?

**Comment by thedanbob** [comment]
I'm seeing this as well: searching while logged in as an admin returns no results while searching logged out works properly. I'm also using docker-compose.

**Comment by pirate** [comment]
What kind of false positives and negatives are you seeing? If you're willing to share screenshots / specific examples of the search queries and bad matches that would help a lot. It could be caused by a number of things, ranging from a bug in the query handling code to using the wrong extractor format for indexing.

**Comment by thedanbob** [comment]
A search for `scar` matches https://www.radiomods.co.nz/kenwood/kenwoodts440.html

`jack` fails to match that page but matches http://tarpn.net/t/faq/faq_networking_on_purpose.html and http://tarpn.net/t/faq/faq_packet_radio.html

`brass`, `dirt`, and `mantra` fail to match https://teddit.net/r/WritingPrompts/comments/5kxe94/wp_you_live_in_a_world_where_each_lie_creates_a/

`dirt` also matches https://nwavguy.blogspot.com/2011/07/o2-headphone-amp.html

I'm running the suggested docker-compose config with PDF, screenshot, DOM, readability, and archive.org saving turned off.

**Comment by pirate** [comment]
Can you try enabling either the `readability` or `mercury` extractor @thedanbob and running `archivebox update --index-only` again? Having at least one article text extractor available will yield the highest quality index, it sometimes struggles when it only indexes raw HTML without the cleaned/extracted text.

**Comment by thedanbob** [comment]
I did have mercury enabled, but I enabled readability as well and reindexed everything. This time a bunch of URLs returned the error `The search backend threw an exception=ERR query_error`. I saw most of the same false positives/negatives, though a few changed:

`jack` no longer matches http://tarpn.net/t/faq/faq_packet_radio.html
`dirt` no longer matches https://nwavguy.blogspot.com/2011/07/o2-headphone-amp.html but does match https://teddit.net/r/WritingPrompts/comments/5pi8t0/pi_everybody_in_the_world_has_a_superpower_that/, a false negative that I didn't catch before. Still has the original false negative.

**Comment by jdcaballerov** [comment]
Thanks @JohnMaguire @thedanbob for taking time to report. Up to now I've noticed a buffer overflow in sonic that might be causing this weird behavior. 

 https://github.com/ArchiveBox/ArchiveBox/pull/625

**Comment by JohnMaguire** [comment]
Given these entries:

![entries](https://user-images.githubusercontent.com/440033/105227297-1e9b0b00-5b2f-11eb-9a9f-b88072c6558b.png)

The following searches return a single result on the public index, but not admin:

- google
- Google

The following searches return on both:

- example
- Example
- b2
- classy
- weblog tool

The following searches return on neither:

- google.com
- example.com

From what I can tell, it is only the Google entry (which was added prior to getting Sonic working) that is failing to return from the Admin search.

**Comment by thedanbob** [comment]
Thanks @jdcaballerov, #625 fixes all of the false negatives I was seeing (and a few I didn't catch at first). I think all of the false positives can be chalked up to sonic's fuzzy search which I wasn't aware of at first.

**Comment by JohnMaguire** [comment]
After building a Docker image from `553c3ca2192e8ce3b42b018ba3bba10e13494748`, running `docker-compose up -d` to create a new container off the new image, and running `docker-compose run archivebox update --index-only`, I am still not getting results for "google" or "Google" in the Admin. It continues to work on the public index.

**Comment by pirate** [comment]
This should be fixed now in v0.5.4, please give it a try. Report back here if you have any issues and I can reopen the ticket.

Please note while the content, title, and tags support full-text search / substring search, URL search must be exact at the moment. This will be improved in a future version.

**Comment by thentoorglan-x** [comment]
I'm still facing this issue.

```
mkdir ~/archivebox && cd ~/archivebox
curl -O 'https://raw.githubusercontent.com/ArchiveBox/ArchiveBox/master/docker-compose.yml'
docker-compose run archivebox init
docker-compose run archivebox --version
docker-compose run archivebox manage createsuperuser
docker-compose up

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3014  100  3014    0     0   6594      0 --:--:-- --:--:-- --:--:--  6594
Starting archivebox_sonic_1 ... done
[i] ArchiveBox v0.5.6: archivebox init
    > /data

[+] Initializing a new ArchiveBox collection in this folder...
    /data
------------------------------------------------------------------

[+] Building archive folder structure...
    √ /data/sources
    √ /data/archive
    √ /data/logs
    √ /data/ArchiveBox.conf

[+] Building main SQL index and running migrations...
    √ /data/index.sqlite3

    Operations to perform:
      Apply all migrations: admin, auth, contenttypes, core, sessions
    Running migrations:
    Applying contenttypes.0001_initial... OK
    Applying auth.0001_initial... OK
    Applying admin.0001_initial... OK
    Applying admin.0002_logentry_remove_auto_add... OK
    Applying admin.0003_logentry_add_action_flag_choices... OK
    Applying contenttypes.0002_remove_content_type_name... OK
    Applying auth.0002_alter_permission_name_max_length... OK
    Applying auth.0003_alter_user_email_max_length... OK
    Applying auth.0004_alter_user_username_opts... OK
    Applying auth.0005_alter_user_last_login_null... OK
    Applying auth.0006_require_contenttypes_0002... OK
    Applying auth.0007_alter_validators_add_error_messages... OK
    Applying auth.0008_alter_user_username_max_length... OK
    Applying auth.0009_alter_user_last_name_max_length... OK
    Applying auth.0010_alter_group_name_max_length... OK
    Applying auth.0011_update_proxy_permissions... OK
    Applying auth.0012_alter_user_first_name_max_length... OK
    Applying core.0001_initial... OK
    Applying core.0002_auto_20200625_1521... OK
    Applying core.0003_auto_20200630_1034... OK
    Applying core.0004_auto_20200713_1552... OK
    Applying core.0005_auto_20200728_0326... OK
    Applying core.0006_auto_20201012_1520... OK
    Applying core.0007_archiveresult... OK
    Applying core.0008_auto_20210105_1421... OK
    Applying sessions.0001_initial... OK

[*] Collecting links from any existing indexes and archive folders...

[*] [2021-03-14 13:48:04] Writing 0 links to main index...
    √ /data/index.sqlite3

------------------------------------------------------------------
[√] Done. A new ArchiveBox collection was initialized (0 links).

    Hint: To view your archive index, run:
        archivebox server  # then visit http://127.0.0.1:8000

    To add new links, you can run:
        archivebox add ~/some/path/or/url/to/list_of_links.txt

    For more usage and examples, run:
        archivebox help
Starting archivebox_sonic_1 ... done
ArchiveBox v0.5.6
Cpython Linux Linux-5.8.0-44-generic-x86_64-with-glibc2.28 x86_64 (in Docker)

[i] Dependency versions:
 √  ARCHIVEBOX_BINARY     v0.5.6          valid     /usr/local/bin/archivebox                                                   
 √  PYTHON_BINARY         v3.9.1          valid     /usr/local/bin/python3.9                                                    
 √  DJANGO_BINARY         v3.1.3          valid     /usr/local/lib/python3.9/site-packages/django/bin/django-admin.py           
 √  CURL_BINARY           v7.64.0         valid     /usr/bin/curl                                                               
 √  WGET_BINARY           v1.20.1         valid     /usr/bin/wget                                                               
 √  NODE_BINARY           v15.8.0         valid     /usr/bin/node                                                               
 √  SINGLEFILE_BINARY     v0.1.14         valid     /node/node_modules/single-file/cli/single-file                              
 √  READABILITY_BINARY    v0.1.0          valid     /node/node_modules/readability-extractor/readability-extractor              
 √  MERCURY_BINARY        v1.0.0          valid     /node/node_modules/@postlight/mercury-parser/cli.js                         
 √  GIT_BINARY            v2.20.1         valid     /usr/bin/git                                                                
 √  YOUTUBEDL_BINARY      v2021.02.04.1   valid     /usr/local/bin/youtube-dl                                                   
 √  CHROME_BINARY         v88.0.4324.146  valid     /usr/bin/chromium                                                           
 √  RIPGREP_BINARY        v0.10.0         valid     /usr/bin/rg                                                                 

[i] Source-code locations:
 √  PACKAGE_DIR           22 files        valid     /app/archivebox                                                             
 √  TEMPLATES_DIR         3 files         valid     /app/archivebox/templates                                                   

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled                                                                              
 -  COOKIES_FILE          -               disabled                                                                              

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data                                                                       
 √  SOURCES_DIR           0 files         valid     ./sources                                                                   
 √  LOGS_DIR              0 files         valid     ./logs                                                                      
 √  ARCHIVE_DIR           0 files         valid     ./archive                                                                   
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf                                                           
 √  SQL_INDEX             196.0 KB        valid     ./index.sqlite3                                                             

Starting archivebox_sonic_1 ... done
[i] ArchiveBox v0.5.6: archivebox manage createsuperuser
    > /data

Username (leave blank to use 'archivebox'): 
Email address: 
Password: 
Password (again): 
Superuser created successfully.
Recreating archivebox_sonic_1 ... done
Recreating archivebox_archivebox_1 ... done
Attaching to archivebox_sonic_1, archivebox_archivebox_1
sonic_1       | thread 'main' panicked at 'cannot read config file: Os { code: 21, kind: Other, message: "Is a directory" }', src/config/reader.rs:24:14
sonic_1       | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
archivebox_sonic_1 exited with code 101
archivebox_1  | [i] ArchiveBox v0.5.6: archivebox server 0.0.0.0:8000
archivebox_1  |     > /data
archivebox_1  | 
archivebox_1  | [+] Starting ArchiveBox webserver...
archivebox_1  |     Hint: The admin username is archivebox
archivebox_1  | 
archivebox_1  | Performing system checks...
archivebox_1  | 
archivebox_1  | System check identified no issues (0 silenced).
archivebox_1  | Django version 3.1.3, using settings 'core.settings'
archivebox_1  | Starting development server at http://0.0.0.0:8000/
archivebox_1  | Quit the server with CONTROL-C.
```

I tried creating config.cfg as described by @JohnMaguire too.

**Comment by thentoorglan-x** [comment]
I'm still facing this issue.

```
mkdir ~/archivebox && cd ~/archivebox
curl -O 'https://raw.githubusercontent.com/ArchiveBox/ArchiveBox/master/docker-compose.yml'
docker-compose run archivebox init
docker-compose run archivebox --version
docker-compose run archivebox manage createsuperuser
docker-compose up

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  3014  100  3014    0     0   6594      0 --:--:-- --:--:-- --:--:--  6594
Starting archivebox_sonic_1 ... done
[i] ArchiveBox v0.5.6: archivebox init
    > /data

[+] Initializing a new ArchiveBox collection in this folder...
    /data
------------------------------------------------------------------

[+] Building archive folder structure...
    √ /data/sources
    √ /data/archive
    √ /data/logs
    √ /data/ArchiveBox.conf

[+] Building main SQL index and running migrations...
    √ /data/index.sqlite3

    Operations to perform:
      Apply all migrations: admin, auth, contenttypes, core, sessions
    Running migrations:
    Applying contenttypes.0001_initial... OK
    Applying auth.0001_initial... OK
    Applying admin.0001_initial... OK
    Applying admin.0002_logentry_remove_auto_add... OK
    Applying admin.0003_logentry_add_action_flag_choices... OK
    Applying contenttypes.0002_remove_content_type_name... OK
    Applying auth.0002_alter_permission_name_max_length... OK
    Applying auth.0003_alter_user_email_max_length... OK
    Applying auth.0004_alter_user_username_opts... OK
    Applying auth.0005_alter_user_last_login_null... OK
    Applying auth.0006_require_contenttypes_0002... OK
    Applying auth.0007_alter_validators_add_error_messages... OK
    Applying auth.0008_alter_user_username_max_length... OK
    Applying auth.0009_alter_user_last_name_max_length... OK
    Applying auth.0010_alter_group_name_max_length... OK
    Applying auth.0011_update_proxy_permissions... OK
    Applying auth.0012_alter_user_first_name_max_length... OK
    Applying core.0001_initial... OK
    Applying core.0002_auto_20200625_1521... OK
    Applying core.0003_auto_20200630_1034... OK
    Applying core.0004_auto_20200713_1552... OK
    Applying core.0005_auto_20200728_0326... OK
    Applying core.0006_auto_20201012_1520... OK
    Applying core.0007_archiveresult... OK
    Applying core.0008_auto_20210105_1421... OK
    Applying sessions.0001_initial... OK

[*] Collecting links from any existing indexes and archive folders...

[*] Writing 0 links to main index...
    √ /data/index.sqlite3

------------------------------------------------------------------
[√] Done. A new ArchiveBox collection was initialized (0 links).

    Hint: To view your archive index, run:
        archivebox server  # then visit http://127.0.0.1:8000

    To add new links, you can run:
        archivebox add ~/some/path/or/url/to/list_of_links.txt

    For more usage and examples, run:
        archivebox help
Starting archivebox_sonic_1 ... done
ArchiveBox v0.5.6
Cpython Linux Linux-5.8.0-44-generic-x86_64-with-glibc2.28 x86_64 (in Docker)

[i] Dependency versions:
 √  ARCHIVEBOX_BINARY     v0.5.6          valid     /usr/local/bin/archivebox                                                   
 √  PYTHON_BINARY         v3.9.1          valid     /usr/local/bin/python3.9                                                    
 √  DJANGO_BINARY         v3.1.3          valid     /usr/local/lib/python3.9/site-packages/django/bin/django-admin.py           
 √  CURL_BINARY           v7.64.0         valid     /usr/bin/curl                                                               
 √  WGET_BINARY           v1.20.1         valid     /usr/bin/wget                                                               
 √  NODE_BINARY           v15.8.0         valid     /usr/bin/node                                                               
 √  SINGLEFILE_BINARY     v0.1.14         valid     /node/node_modules/single-file/cli/single-file                              
 √  READABILITY_BINARY    v0.1.0          valid     /node/node_modules/readability-extractor/readability-extractor              
 √  MERCURY_BINARY        v1.0.0          valid     /node/node_modules/@postlight/mercury-parser/cli.js                         
 √  GIT_BINARY            v2.20.1         valid     /usr/bin/git                                                                
 √  YOUTUBEDL_BINARY      v2021.02.04.1   valid     /usr/local/bin/youtube-dl                                                   
 √  CHROME_BINARY         v88.0.4324.146  valid     /usr/bin/chromium                                                           
 √  RIPGREP_BINARY        v0.10.0         valid     /usr/bin/rg                                                                 

[i] Source-code locations:
 √  PACKAGE_DIR           22 files        valid     /app/archivebox                                                             
 √  TEMPLATES_DIR         3 files         valid     /app/archivebox/templates                                                   

[i] Secrets locations:
 -  CHROME_USER_DATA_DIR  -               disabled                                                                              
 -  COOKIES_FILE          -               disabled                                                                              

[i] Data locations:
 √  OUTPUT_DIR            6 files         valid     /data                                                                       
 √  SOURCES_DIR           0 files         valid     ./sources                                                                   
 √  LOGS_DIR              0 files         valid     ./logs                                                                      
 √  ARCHIVE_DIR           0 files         valid     ./archive                                                                   
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf                                                           
 √  SQL_INDEX             196.0 KB        valid     ./index.sqlite3                                                             

Starting archivebox_sonic_1 ... done
[i] ArchiveBox v0.5.6: archivebox manage createsuperuser
    > /data

Username (leave blank to use 'archivebox'): 
Email address: 
Password: 
Password (again): 
Superuser created successfully.
Recreating archivebox_sonic_1 ... done
Recreating archivebox_archivebox_1 ... done
Attaching to archivebox_sonic_1, archivebox_archivebox_1
sonic_1       | thread 'main' panicked at 'cannot read config file: Os { code: 21, kind: Other, message: "Is a directory" }', src/config/reader.rs:24:14
sonic_1       | note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
archivebox_sonic_1 exited with code 101
archivebox_1  | [i] ArchiveBox v0.5.6: archivebox server 0.0.0.0:8000
archivebox_1  |     > /data
archivebox_1  | 
archivebox_1  | [+] Starting ArchiveBox webserver...
archivebox_1  |     Hint: The admin username is archivebox
archivebox_1  | 
archivebox_1  | Performing system checks...
archivebox_1  | 
archivebox_1  | System check identified no issues (0 silenced).
archivebox_1  | Django version 3.1.3, using settings 'core.settings'
archivebox_1  | Starting development server at http://0.0.0.0:8000/
archivebox_1  | Quit the server with CONTROL-C.
```

I tried creating config.cfg as described by @JohnMaguire too.

**Comment by thentoorglan-x** [comment]
Ubuntu 20.04 LTS @pirate

**Comment by JohnMaguire** [comment]
@thentoorglan-x What issue are you experiencing? Your logs do not include the error reported in the original post here (which is regarding a missing config.cfg). Your logs appear as though the service started up correctly. I'd advise you file a new ticket clearly describing what you're experiencing versus what you expected.

**Comment by erob8** [comment]
Also had the issue described initially in this thread and recently by @thentoorglan-x . I think the sonic section in docker-compose.yaml file should have its volume updated to

```
        volumes:
            - ./etc/sonic/config.cfg:/etc/sonic.cfg:ro
            - ./data/sonic:/var/lib/sonic/store
```

This syncs the config file to where it is placed in the wget command

 `wget https://raw.githubusercontent.com/ArchiveBox/ArchiveBox/master/etc/sonic/config.cfg -O etc/sonic/config.cfg` 

from the wiki page https://github.com/ArchiveBox/ArchiveBox/wiki/Docker#setup

**Comment by pirate** [comment]
I updated it in the docker-compose.yml file on `dev` yesterday, take a look at the comment section above the sonic block for instructions. Will update the wiki now too.

**Comment by asitemade4u** [comment]
I have tried the last version in `dev` and face the same issue as previously described.
IMO the question is WHERE to save the Sonic `config.cfg` file.
I am using a docker installation nested within a Proxmox container -- it works very well and I have currently more than  docker servers working that way...
So, I tried to download the config file in:
- the enclosing Proxmox container, at the root `/config.cfg`
- the root level of the Archivebox container
None worked.
Are you sure the path to the config file should be, in the `docker-compose.yaml` file `./config.cfg`?
